addon/components/g-map-polyline.js

```diff
@@ -0,0 +1,205 @@
+    if (isPresent(event)) {
+      polyline.addListener(event, () => infowindow.open());
+    }
+  },
+
+  attachCloseInfowindowEvent(polyline, infowindow, event) {
+    if (isPresent(event)) {
+      polyline.addListener(event, () => infowindow.close());
+    }
+  },
+
+  attachTogglingInfowindowEvent(polyline, infowindow, event) {
+    if (isPresent(event)) {
+      polyline.addListener(event, () => {
+        if (infowindow.get('isOpen')) {
+          infowindow.close();
+        } else {
+          infowindow.open();
+        }
+      });
+    }
+  },
+
+  unsetPolylineFromMap() {
+    const polyline = this.get('polyline');
+    if (isPresent(polyline)) {
+      polyline.setMap(null);
+    }
+  },
+
+  mapWasSet: observer('map', function() {
+    run.once(this, 'setMap');
+  }),
+
+  setMap() {
+    const map = this.get('map');
+    const polyline = this.get('polyline');
+
+    if (isPresent(polyline) && isPresent(map)) {
+      polyline.setMap(map);
+    }
+  },
+
+  setPath() {
+    const polyline = this.get('polyline');
+    const coordinates = this.get('coordinates');
+    const options = compact(this.getProperties(allowedPolylineOptions));
+    var coordArray = [];
+    coordinates.forEach(function(coordinate){
+      const coord = coordinate.get('coordinate');
+      if (isPresent(coord)) {
+        coordArray.push(coord);
+      }
+    });
+    Ember.Logger.log(coordArray);
+    // assert('Must have at least two valid coordinates in {{#g-map-polyline}} component', coordArray.length > 1);
+    if (coordArray.length > 1 && isPresent(polyline) && isPresent(coordinates)) {
+      polyline.setPath(coordArray);
+    }
+    if (isPresent(polyline) && isPresent(options)) {
+      polyline.setOptions(options);
+    }
+  },
+
+  iconChanged: observer('icon', function() {
+    run.once(this, 'setIcon');
+  }),
+
+  setOnClick() {
+    const polyline = this.get('polyline');
+    if (isPresent(polyline)) {
+      polyline.addListener('click', () => this.sendOnClick());
+    }
+  },
+
+  labelChanged: observer('label', function() {
+    run.once(this, 'setLabel');
+  }),
+
+  setLabel() {
+    const polyline = this.get('polyline');
+    const label = this.get('label');
+
+    if (isPresent(polyline) && isPresent(label)) {
+      polyline.setLabel(label);
+    }
+  },
+
+  titleChanged: observer('title', function() {
+    run.once(this, 'setTitle');
+  }),
+
+  setTitle() {
+    const polyline = this.get('polyline');
+    const title = this.get('title');
+
+    if (isPresent(polyline) && isPresent(title)) {
+      polyline.setTitle(title);
+    }
+  },
+
+  sendOnClick() {
+    const { onClick } = this.attrs;
+    const mapContext = this.get('mapContext');
+    const group = this.get('group');
+
+    if (typeOf(onClick) === 'function') {
+      onClick();
+    } else {
+      this.sendAction('onClick');
+    }
+
+    if (isPresent(group)) {
+      mapContext.groupPolylineClicked(this, group);
```

Comment:
I can't find that method in map, however it's nice idea to have the logic similar to markers auto-closing in groups. Could you add this logic for polylines as well?